Fix flaky test in TestPlannedReparenter_performGracefulPromotion

#### Overview of the Issue

Flaky test in PlannedReparenter unit tests, specifically a test case in `performGracefulPromotion` around timeouts.

Somehow, the way we're setting up the test case doesn't always guarantee a timeout in the parent context, which was the intention:

```
--- FAIL: TestPlannedReparenter_performGracefulPromotion (0.00s)
    --- FAIL: TestPlannedReparenter_performGracefulPromotion/demotion_succeeds_but_parent_context_times_out (0.01s)
        planned_reparenter_test.go:1446: 
            	Error Trace:	planned_reparenter_test.go:1446
            	Error:      	An error is expected but got nil.
            	Test:       	TestPlannedReparenter_performGracefulPromotion/demotion_succeeds_but_parent_context_times_out
FAIL
```

#### Reproduction Steps

Run the test a few (many?) times until you get a failure.

--- a/go/vt/vtctl/reparentutil/planned_reparenter_test.go
+++ b/go/vt/vtctl/reparentutil/planned_reparenter_test.go
@@ -1100,7 +1100,7 @@ func TestPlannedReparenter_performGracefulPromotion(t *testing.T) {
 					"zone1-0000000200": nil,
 				},
 				WaitForPositionPostDelays: map[string]time.Duration{
-					"zone1-0000000200": time.Millisecond * 5,
+					"zone1-0000000200": time.Second * 1,
 				},
 				WaitForPositionResults: map[string]map[string]error{
 					"zone1-0000000200": {
@@ -1349,7 +1349,7 @@ func TestPlannedReparenter_performGracefulPromotion(t *testing.T) {
 					},
 				},
 				PromoteReplicaPostDelays: map[string]time.Duration{
-					"zone1-0000000200": time.Millisecond * 20, // 2x the parent context timeout
+					"zone1-0000000200": time.Millisecond * 100, // 10x the parent context timeout
 				},
 				PromoteReplicaResults: map[string]struct {
 					Result string

